tests/views/PageSpotDetail.spec.js

```diff
@@ -0,0 +1,175 @@
+                        toast: {
+                            open: vi.fn(),
+                        },
+                    },
+                },
+                stubs: {
+                    LoaderModal: {
+                        template: '<div class="loader-modal"></div>',
+                    },
+                    TemplateSpotDetail: {
+                        template: '<div class="template-spot-detail"></div>',
+                    },
+                },
+            },
+        });
+
+    beforeEach(() => {
+        mockPush.mockClear();
+
+        actions = {
+            getSpotDetails: vi.fn().mockResolvedValue(),
+            updateAvailability: vi.fn().mockResolvedValue(),
+            updateLastCallDate: vi.fn().mockResolvedValue(),
+            updateRemark: vi.fn().mockResolvedValue(),
+        };
+
+        store = createStore({
+            modules: {
+                sdp: {
+                    namespaced: true,
+                    state: {
+                        loading: false,
+                        title: 'Spot Detail',
+                    },
+                    actions,
+                },
+                searchPortal: {
+                    namespaced: true,
+                    state: {
+                        activeTab: 1,
+                    },
+                    actions: {
+                        updateActiveTab: vi.fn(),
+                        updateSOLatLngInput: vi.fn(),
+                    },
+                },
+                map: {
+                    namespaced: true,
+                    mutations: {
+                        'update-user-location': vi.fn(),
+                    },
+                },
+            },
+        });
+
+        Object.defineProperty(global.navigator, 'geolocation', {
+            value: {
+                getCurrentPosition: vi.fn(),
+            },
+            configurable: true,
+        });
+    });
+
+    it('shows loader when page is loading', () => {
+        store.state.sdp.loading = true;
+        const wrapper = mountComponent();
+
+        expect(wrapper.find('.loader-modal').exists()).toBe(true);
+        expect(wrapper.find('.template-spot-detail').exists()).toBe(false);
+    });
+
+    it('renders spot detail template when loading is false', () => {
+        const wrapper = mountComponent();
+        expect(wrapper.find('.template-spot-detail').exists()).toBe(true);
+    });
+
+    it('fetches spot details on mount with encoded spotId', async () => {
+        mountComponent();
+        await flushPromises();
+
+        expect(actions.getSpotDetails).toHaveBeenCalledWith(expect.anything(), {
+            spotId: 'SPOT%23123',
+        });
+    });
+
+    it('requests user location on mount', async () => {
+        mountComponent();
+        await flushPromises();
+
+        expect(navigator.geolocation.getCurrentPosition).toHaveBeenCalled();
+    });
+
+    it('updates user location on geolocation success', () => {
```

Comment:
This only checks if the mutation exists, not if it was called with correct values. Consider using a spy:
```
it('updates user location on geolocation success', () => {
    const commitSpy = vi.spyOn(store, 'commit');
    const wrapper = mountComponent();
    wrapper.vm.onGeoSuccess({
        coords: { latitude: 28.61, longitude: 77.23 },
    });
    expect(commitSpy).toHaveBeenCalledWith('map/update-user-location', [28.61, 77.23]);
});
```